parachain/pallets/erc20-app/src/lib.rs

```diff
@@ -159,8 +162,8 @@ pub mod pallet {
 			let asset_id = AssetId::Token(token);
 			T::Assets::deposit(asset_id, &recipient, amount)?;
 
-			if let Some(id) = para_id {
-				T::XcmReserveTransfer::reserve_transfer(origin, asset_id, id, &recipient, amount)?;
+			if let Some(destination) = destination {
+				T::XcmReserveTransfer::reserve_transfer(asset_id, &recipient, amount, destination)?;
```

Comment:
My comment in `eth-app/lib.rs` also applies here. We should be ignoring any errors returned by `reserve_transfer`.

Reply:
ye though probably need some log/event that can be detected by the frontend or some third party for clear ux for the end user